yaksh/models.py

```diff
@@ -720,8 +723,43 @@ class CourseStatus(models.Model):
     course = models.ForeignKey(Course)
     user = models.ForeignKey(User)
     grade = models.CharField(max_length=255, null=True, blank=True)
-    total_marks = models.FloatField(default=0.0)
+    percentage = models.FloatField(default=0.0)
 
+    def get_grade(self):
+        return self.grade
+
+    def set_grade(self):
+        if self.is_course_complete():
+            self.calculate_percentage()
+            if self.course.grading_system is None:
+                grading_system = GradingSystem.objects.get(name='default')
+            else:
+                grading_system = self.course.grading_system
+            grade = grading_system.get_grade(self.percentage)
+            self.grade = grade
+
+    def calculate_percentage(self):
+        if self.is_course_complete():
+            quizzes = self.course.get_quizzes()
+            total_weightage = 0
+            sum = 0
+            for quiz in quizzes:
+                total_weightage += quiz.weightage
+                marks = AnswerPaper.objects.get_user_best_of_attempts_marks(
+                        quiz, self.user.id, self.course.id)
+                out_of = quiz.questionpaper_set.first().total_marks
+                sum += (marks/out_of)*quiz.weightage
+            self.percentage = (sum/total_weightage)*100
+
```

Comment:
I guess you have not added tests for `calculate_percentage` and `set_grade`

Reply:
Yes because CourseStatus tests are not present in model tests and I am adding them but few problems I am facing currently so have mention in the issue that tests will be written.
Though will write tests for the methods.